### ISSUE TITLE
new 1.6 pep8 release makes some branches red

### ISSUE THREAD

**Issue by lepistone** [OCA/maintainer-quality-tools#164]
pep8 adds some new checks by default: https://github.com/jcrocholl/pep8/blob/master/CHANGES.txt#L25

The first I found was E402 (imports must be at top of file) and it seemed reasonable and easy to fix in OCA/sale-workflow#108, but there may be others. So we could:

1 fix any other individual issue like I did for sale-workflow
2 ignore the new checks that make some of our branches red
3 keep an old pep8 version

I suggest 1 if easy to fix and reasonable, otherwise 2. What do you think?

**Comment by pedrobaeza** [comment]
I think 1 is suitable and very easy to fix. I have done it for account-financial-tools also: https://github.com/OCA/account-financial-tools/pull/143

**Comment by dreispt** [comment]
I have just seen [removing lambdas](https://github.com/OCA/management-system/pull/58) because of E731,
To me that doesn't make sense and the check should be disabled.

**Comment by lepistone** [comment]
@dreispt I think the use we do of lambdas in odoo is pep8-legal, and the pep8 tool should be ok with it in the latest version. See the discussion on https://github.com/jcrocholl/pep8/issues/311.

TL;DR : it is **direct assignment** which is pep8-forbidden: `f = lambda x: 2 * x`, because you can replace it with `def f(x): return 2 * x` and you get better introspection of `f`. 

Instead, our use which is `Field(parameter=lambda x: 2 * x)` should be recognized by the pep8 tool as legal.

But maybe it isn't... if that is the case, we can of course disable the check.

Anyway I agree with you @dreispt that one way or the other we can keep our lambdas in field definitions.

**Comment by dreispt** [comment]
In the [case](https://github.com/OCA/management-system/pull/58/files#diff-1d9a94ba8a20cc5c6aa2877a466dfbaaR42) I mentioned lambdas are being used in the default dict.
I think that is an assigned and so triggers the check.

**Comment by lepistone** [comment]
@dreispt I tested with `pep8` 1.6.2, the latest version. It is as I said:
- a **direct assignment** of a lambda to a variable, like `g = lambda x: 2 * x` raises E731.
- something like:

```python
_defaults = {
    'company_id': (
        lambda self, cr, uid, c:
        self.pool.get('res.users').browse(cr, uid, uid, c).company_id.id),
}
```
does not raise E731, because it is not a direct assignment to a variable.

So I think with the latest `pep8` we don't have anything to do. Do you have an example where it's not like that? Thanks!

**Comment by gurneyalex** [comment]
I'm tempted to close this issue, unless people object?

**Comment by yvaucher** [comment]
I think this has been addressed in most repo. So no objection to close it.
